packages/runtime/src/runtime.ts

```diff
@@ -87,6 +88,10 @@ export class Runtime {
       throw new RuntimeError(RUNTIME_ERRORS.GENERAL.ACCOUNT_DOES_NOT_EXIST,
         { address: address, line: lineNumber });
     }
+    if (a.address !== address){
```

Comment:
Interesting, I'm curious why this is possible. If this happens then we have a bug.

Reply:
I agree that a test is needed. Sorry that I forgot to explain. (I cut my comment to paste it here but forgot, I'll add the part below to the title)

## Why 
The point of this PR(my intention) was the calling pattern, although it's functionally correct: 

This `assertAccountDefined` can be done before TS compile, but maybe for JS it's needed. I saw two passed args (`address` and  `account`) when I was with TS. This would be a little bit confusing, for making users think the address and the account are related. 

## Example
E.g. For `assertAccountDefined(acc1.address,acc2)`, naming like `assertItsGood`. It's hard to see if the output would be 
1. boolean
2. an acc, but unclear whether the output would be acc1 or acc2
3. does it asserts acc1==acc2 (at least with same addr)

This PR resolves the ambiguity of (2) and (3)

Here it works like a filter, maybe renaming it would be better?

Reply:
Ok

Reply:
Let's keep the same name.